python/ngraph/ops.py

```diff
@@ -520,6 +520,50 @@ def broadcast_to(node, new_shape, axis=None, name=None):
     return Broadcast(node, Shape(new_shape), get_broadcast_axes(new_shape, node.shape, axis))
 
 
+@nameable_op
+def gemm(A,                      # type: Node
+         B,                      # type: Node
+         C,                      # type: Node
+         alpha,                  # type: ScalarData
+         beta,                   # type: ScalarData
+         transA,                 # type: bool
+         transB,                 # type: bool
+         name=None,              # type: str
+         ):
+    # type: (...) -> Node
+    r"""Perform General matrix-matrix multiplication on input tensors A, B and C.
+
+    Computes:
+
+    .. math:: Y = alpha\cdot A'\cdot B' +  beta\cdot C
+
+    :code:`A'`: The transpose of matrix :code:`A` with shape (M, K),
+                if :code:`transA` is :code:`True`, otherwise :code:`A` with shape (K, N).
+
+    :code:`B'`: The transpose of matrix :code:`B` with shape (K, N),
+                if :code:`transB` is :code:`True`, otherwise :code:`B` with shape (N, K).
+
+    :code:`C`: Matrix broadcastable to shape (M, N).
+
+    :code:`Y`: Matrix with shape (M, N).
+
+    For more information refer to:
```

Comment:
GEMM, as the name suggests is a very general algorithm. I don't think we need to link to any publication for it.

Also, avoid deep-linking into PDF files, let's use links to Arxiv like this: https://arxiv.org/abs/1709.03395